tools/functional-tester/etcd-tester/tester.go

```diff
@@ -131,53 +139,19 @@ func (tt *tester) runLoop() {
 	}
 }
 
-func (tt *tester) compact(rev int64, timeout time.Duration) error {
-	plog.Printf("%s compacting storage (current revision %d, compact revision %d)", tt.logPrefix(), tt.currentRevision, rev)
-	if err := tt.cluster.compactKV(rev, timeout); err != nil {
-		plog.Printf("%s compactKV error (%v)", tt.logPrefix(), err)
-		if cerr := tt.cleanup(); cerr != nil {
-			return fmt.Errorf("%s, %s", err, cerr)
-		}
-		return err
-	}
-	plog.Printf("%s compacted storage (compact revision %d)", tt.logPrefix(), rev)
-
-	plog.Printf("%s checking compaction (compact revision %d)", tt.logPrefix(), rev)
-	if err := tt.cluster.checkCompact(rev); err != nil {
-		plog.Printf("%s checkCompact error (%v)", tt.logPrefix(), err)
-		if cerr := tt.cleanup(); cerr != nil {
-			return fmt.Errorf("%s, %s", err, cerr)
-		}
-		return err
+func (tt *tester) checkCurrentRevision() error {
+	revs, _, err := tt.cluster.getRevisionHash()
+	for _, rev := range revs {
```

Comment:
probably just break it out at the first time and also add a comment for why you need to do this.